### PR TITLE
JDNS: fixes due to static code analysis

### PR DESCRIPTION
**Pull request by strahlex**
- reduced scope of variables
- added initialization member variables
- removed C-style typecasts from QJDNS
- using enum instead of integer

### CODE REVIEW

File: src/jdns/jdns_packet.c [hunk 1]
```diff
@@ -310,6 +310,7 @@ int jdns_packet_name_isvalid(const unsigned char *name, int size)
 	at = 0;
 	while(1)
 	{
+        int len;
```
Comment (drizt) [inline src/jdns/jdns_packet.c]: bad indentation

File: src/jdns/jdns_sys.c [hunk 2]
```diff
@@ -671,6 +669,10 @@ static jdns_dnsparams_t *dnsparams_get_unixfiles()
 		return params;
 	while(1)
 	{
+		 int n;
```
Comment (drizt) [inline src/jdns/jdns_sys.c]: bad indentation

**Comment by drizt** [comment]
Pleade don't make a new pull request. Just update current. Used `git commit --amend` for this.

File: src/jdns/jdns_packet.c (hunk 1 above)
Comment (strahlex) [inline src/jdns/jdns_packet.c]: nope, exactly like it should be

File: src/jdns/jdns_sys.c (hunk 2 above)
Comment (strahlex) [inline src/jdns/jdns_sys.c]: same here

File: src/jdns/jdns_packet.c (hunk 1 above)
Comment (drizt) [inline src/jdns/jdns_packet.c]: use two tabs instead of spaces.

**Comment by strahlex** [comment]
I would, but there is no force option for git subtree.

Ivan Romanov writes:

> Pleade don't make a new pull request. Just update current. Used `git commit --amend` for this.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/psi-im/jdns/pull/5#issuecomment-85659708

-- 
Alexander Rössler

File: src/jdns/jdns_sys.c (hunk 2 above)
Comment (drizt) [inline src/jdns/jdns_sys.c]: There extra a space before `int`

**Comment by strahlex** [comment]
Hope I found all indentations. Sorry, but QtCreator does not make clear what are spaces and what are tabs. Switched to Emacs to clean it up, but have overseen a few.

**Comment by drizt** [comment]
`git diff` shows indentations.

**Comment by strahlex** [comment]
But not if the indentations are spaces or tabs. Only few editors and
tools do.

Ivan Romanov writes:

**Pull request merged**